src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -71,18 +71,17 @@ pub fn flashback_to_version_read_write(
 // `CF_LOCK`.
 pub fn rollback_locks(
     txn: &mut MvccTxn,
-    reader: &mut SnapshotReader<impl Snapshot>,
+    snapshot: impl Snapshot,
     key_locks: Vec<(Key, Lock)>,
 ) -> TxnResult<Option<Key>> {
     for (key, lock) in key_locks {
         if txn.write_size() >= MAX_TXN_WRITE_SIZE {
             return Ok(Some(key));
         }
         // To guarantee rollback with start ts of the locks
-        reader.start_ts = lock.ts;
         rollback_lock(
             txn,
-            reader,
+            &mut SnapshotReader::new(lock.ts, snapshot.clone(), false),
```

Comment:
It's not necessary to create a `SnapshotReader` for each key, in fact, you can just create one outside the loop.